### PR TITLE
Release 1.0.4

### PR DESCRIPTION
**Pull request by vhrychenko**
- Developer(s): @vhrychenko

- Ticket: https://spryker.atlassian.net/browse/ECO-2707

#### Please confirm

- [x] No new OS components - or they have been approved by legal department

#### Documentation

- [x] Functional documentation provided or in progress?
- [x] Integration guide for projects provided or not needed?
- [x] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   AfterPay              | patch                 |              |

#### Release Notes

Fixed argument type mismatch if a reservation was not successful

-----------------------------------------

##### Change log

Bugfixes

- Adjusted `PaymentAuthorizeWriter::save()` so it might get nullable parameters (`$idReservation` and `$idCheckout`) in case a reservation was not successful. A parameters order was changed. 
- Adjusted `PaymentWriter::save()` so it might get nullable `$idReservation` parameter in case a reservation was not successful. A parameters order was changed. 
- Allow new core majors

### CODE REVIEW

File: src/SprykerEco/Zed/AfterPay/Business/Payment/PaymentWriter.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ public function __construct(
      *
      * @return void
      */
-    public function setIdReservationByIdSalesOrder(string $idReservation, int $idSalesOrder): void
+    public function setIdReservationByIdSalesOrder(int $idSalesOrder, string $idReservation = null): void
```
Comment (gechetspr) [inline src/SprykerEco/Zed/AfterPay/Business/Payment/PaymentWriter.php]: `$idReservation` is nullable

File: src/SprykerEco/Zed/AfterPay/Business/Payment/PaymentWriterInterface.php [hunk 2]
```diff
@@ -15,7 +15,7 @@ interface PaymentWriterInterface
      *
      * @return void
      */
-    public function setIdReservationByIdSalesOrder(string $idReservation, int $idSalesOrder): void;
+    public function setIdReservationByIdSalesOrder(int $idSalesOrder, string $idReservation = null): void;
```
Comment (gechetspr) [inline src/SprykerEco/Zed/AfterPay/Business/Payment/PaymentWriterInterface.php]: same

File: src/SprykerEco/Zed/AfterPay/Business/Payment/Transaction/Authorize/PaymentAuthorizeWriter.php [hunk 3]
```diff
@@ -32,7 +32,7 @@ public function __construct(AfterPayQueryContainerInterface $afterPayQueryContai
      *
      * @return void
      */
-    public function save(string $orderReference, string $idReservation, string $idCheckout): void
+    public function save(string $orderReference, string $idReservation = null, string $idCheckout = null): void
```
Comment (gechetspr) [inline src/SprykerEco/Zed/AfterPay/Business/Payment/Transaction/Authorize/PaymentAuthorizeWriter.php]: same

File: src/SprykerEco/Zed/AfterPay/Business/Payment/Transaction/Authorize/PaymentAuthorizeWriterInterface.php [hunk 4]
```diff
@@ -16,5 +16,5 @@ interface PaymentAuthorizeWriterInterface
      *
      * @return void
      */
-    public function save(string $orderReference, string $idReservation, string $idCheckout): void;
+    public function save(string $orderReference, string $idReservation = null, string $idCheckout = null): void;
```
Comment (gechetspr) [inline src/SprykerEco/Zed/AfterPay/Business/Payment/Transaction/Authorize/PaymentAuthorizeWriterInterface.php]: same

**Review by gechetspr — CHANGES_REQUESTED** [review]
Please specify in change logs, that param order for method was changed

**Review by gechetspr — APPROVED** [review]

File: tests/_data/after-pay.databuilder.xml [hunk 5]
```diff
@@ -93,7 +93,7 @@
 
     <transfer name="Item" >
         <property name="sku" dataBuilderRule="isbn10" />
-        <property name="unitGrossPrice" dataBuilderRule="randomNumber(5)"  />
+        <property name="unitGrossPrice" dataBuilderRule="randomFloat(2)"  />
```
Comment (alex-galych) [inline tests/_data/after-pay.databuilder.xml]: This should be fixed

**Review by alex-galych — APPROVED** [review]

**Pull request merged**